### PR TITLE
Fix facet tag removal for multiple search terms

### PR DESCRIPTION
**Pull request by vanitabarrett**
Trello: https://trello.com/c/udoZSC9q/331-bug-fix-facet-tag-removal

This commit fixes a bug with facet tag removal for text searches. Facet tag removal was always removing the first instance of that keyword, even if it was part of another word, e.g: "therefore the". In this case, removing "the" would result in "refore the". This commit fixes that.

Approaches:

- Surrounding the keyword with a space on both sides before looking it up in the input value. However, the first and last value won't have spaces on both sides
- Adding an index to the facet tag so it can be looked up and removed by index. I got this partially implemented by adding a data attribute in KeywordFacet. However, this seemed to disappear each time a facet tag was removed. To get this solution working, I think we'd need to spend more time looking into how the facet tags are generated and/or refreshed.
- Splitting the input value by space and looping through it, replacing the current word with "" when it matched the facet tag being removed. However, this meant if a keyword existed in the search term multiple times, it was removed in all instances.
- **Implemented approach:** expanded on the above approach by exiting the loop early when the keyword was found. This means if a keyword exists multiple times in a search time, it is only removed once.

Given the following search:
Screen Shot 2019-03-21 at 16 19 19

trying to remove "the":

## Before
Screen Shot 2019-03-21 at 16 19 12

## After
Screen Shot 2019-03-22 at 10 46 34

Heroku: https://finder-frontend-pr-988.herokuapp.com/find-eu-exit-guidance-business

### CODE REVIEW

**Comment by vanitabarrett** [comment]
**Note: looking into this again as it still doesn't work as expected**

**Comment by vanitabarrett** [comment]
@andysellick Could you please take another look? I've pushed a solution which exists the loop early once a keyword is found. I think it works... 🤞 Other tried approaches documented in the PR description.

File: app/assets/javascripts/modules/remove-filter.js [hunk 1]
```diff
@@ -33,7 +33,18 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
...
+        var splitKeywords = currentVal.split(" ");
+
+        for (var i = 0; splitKeywords.length; i += 1) {
+          if (splitKeywords[i].toString() === removeFilterValue.toString()) {
+            splitKeywords.splice(i, 1);
+            break;
+          }
+        }
+
+        var newVal = splitKeywords.join(" ")
```
Comment (andysellick) [inline app/assets/javascripts/modules/remove-filter.js]: Nitpick: semi-colon.

File: app/assets/javascripts/modules/remove-filter.js [hunk 2]
```diff
@@ -33,7 +33,18 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
...
+
+        $input.val(newVal.trim()).trigger({
```
Comment (andysellick) [inline app/assets/javascripts/modules/remove-filter.js]: Nitpick: I'd add the `trim` onto the end of line 45, just to make this line only about setting a value.

File: app/assets/javascripts/modules/remove-filter.js [hunk 3]
```diff
@@ -33,7 +33,18 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
         $input.trigger('change');
       }
       else if (inputType == 'text' || inputType == 'search') {
-        $input.val(currentVal.replace(removeFilterValue, '').replace(/\s+/g,' ').trim()).trigger({
+        var splitKeywords = currentVal.split(" ");
+
+        for (var i = 0; splitKeywords.length; i += 1) {
```
Comment (andysellick) [inline app/assets/javascripts/modules/remove-filter.js]: Interesting, I'd write this as `i++`, I've not seen it done this way before. Works fine though.

Comment (vanitabarrett) [reply app/assets/javascripts/modules/remove-filter.js]: Doh, I've got too used to Ruby not having `++` or `--`.

**Pull request merged**